kong/clustering/services/sync/rpc.lua

```diff
@@ -413,12 +402,27 @@ function sync_once_impl(premature, retry_count)
 
   -- retry if the version is not updated
   retry_count = retry_count or 0
+
   if retry_count > MAX_RETRY then
     ngx_log(ngx_ERR, "sync_once retry count exceeded. retry_count: ", retry_count)
     return
   end
 
-  return start_sync_once_timer(retry_count + 1)
+  -- we do not count a timed out sync. just retry
+  if err ~= "timeout" then
```

Comment:
Now `timeout` option is `0`:
```
local SYNC_MUTEX_OPTS = { name = "get_delta", timeout = 0, }
```
So how the error `timeout` will happen?

Reply:
@chronolaw This is a locked operation, where `timeout = 0` means it will fail instantly if the lock can not be acquired. By tolerating timeout in this case, we allow `sync_once` to actually retry instead of giving up when another coroutine holds the lock

Reply:
Do  you mean that it will always get a `timeout` error?

Reply:
@chronolaw No. When the callback fails or errors out it will emit a different error. That is what we count for a real error

Reply:
I.e, if we have no error but the version still does not match, we count it as 1 trial; if it's a real failure (internal or external), we still count it as 1 trial. This way we prevent it from looping forever; but meanwhile we do not want a running coroutine to block the `sync_once` call and make it quickly exhaust all the chances, so we do not count timeouts(lock failure)